monitoring/grafana/dashboards/tests/util.py

```diff
@@ -0,0 +1,68 @@
+import json
+from pathlib import Path
+from typing import Any, Dict, Tuple, Union
+
+from termcolor import cprint
+
+UNITS = ['ms', 's', 'm', 'h', 'd', 'w', 'y']
+
+
+def resolve_time_and_unit(time: str) -> Union[Tuple[int, str], Tuple[None, None]]:
+    """
+    Divide time with its unit and return a tuple like (10, 'm')
+    Return None if its and invalid prometheus time
+    Valid units are inside UNITS.
+    """
+    if time[-1] in UNITS:
+        return int(time[:-1]), time[-1]
+    if time[-2:] in UNITS:
+        return int(time[:-2]), time[-2:]
+    return None, None
+
+
+def get_dashboards_data() -> Dict[str, Any]:
+    data: Dict[str, Any] = {'queries': {}, 'variables': {}, 'stats': {}}
+    for file in Path(__file__).parent.parent.glob('*.json'):
+        with open(file, 'r') as f:
+            dashboard_data = json.load(f)
+            data['stats'][str(file)] = {'total': 0, 'tested': 0}
+            add_dashboard_queries(data, dashboard_data, str(file))
+            add_dashboard_variables(data, dashboard_data)
+    return data
+
+
+def add_dashboard_queries(data: Dict[str, Any], dashboard_data: Dict[str, Any], path: str) -> None:
+    """
+    Grafana panels can have more than one target/query, in order to identify each
+    query in the panel we append the "legendFormat" of the target to the panel name.
+    format: panel_name-legendFormat
+    """
+    if 'panels' not in dashboard_data:
+        return
+    for panel in dashboard_data['panels']:
+        if (
+                'title' in panel
+                and 'targets' in panel
+                and len(panel['targets']) > 0
+                and 'expr' in panel['targets'][0]
+        ):
+            for target in panel['targets']:
+                title = panel['title']
+                legend_format = target['legendFormat'] if 'legendFormat' in target else ""
+                query_id = title + '-' + legend_format
```

Comment:
If you're parsing this from 2 different input fields (`panel_name` and `legend`... and now probably also a `ref_id` letter or array `index`), why not saving those in nested dirs?:
```py
data['queries'][panel_name][legend][ref_id]
```
That brings a challenge, which is automatically creating a nested dir on access (could be solved with a `defaultdict` whose default items are `dicts`) and also how to deal with missing keys (could be solved with a Javascript-like dot behaviour).

A library that does both things is [python-box](https://github.com/cdgriffith/Box/wiki/Types-of-Boxes):
```py
data['queries'] = Box(default_box=True, box_dots=True)

# for storing data you can use both the dot-notation or the regular brackets (it'll autocreate the missing keys)
data['queries'].Pool_Overview = "..."
data['queries'].Pool_Details.RW_Writes = "..."
data['queries']["Pool_Details"]["RW_Reads"] = "..."
data['queries']["Pool_Details.RW_Reads"] = "..."  # this works too

# for retrieving data
data['queries'][table][legend][id] # this will create any missing key

# to void creating new keys you need to use the `.get(key)`
data['queries'].get(f'{table}.{legend}.{id}')

# in order to make this easier you can always assume a default value for legend or id when not defined
```